### ISSUE TITLE
pyabf compatibility with pyInstaller

### ISSUE THREAD

**Issue by SebastianOltmanns** [swharden/pyABF#109]
Hello,

I use pyABF in my project and I am trying to create an executable with [pyInstaller](https://pyinstaller.readthedocs.io/en/stable/index.html). I have problems running the executable.

### The problem

For the purpose of this issue, let me describe a very simple test project. The following file is called `test.py`: 

```python
from pyabf import ABF

print('hello')
print(ABF)
print('bye')
```

Using the newest version of pyInstaller, I can run `pyinstaller test.py --onefile` and an executable is created. When I run this executable, the following error message is shown:

```
 Traceback (most recent call last):
  File "test.py", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "c:\users\sebu\appdata\local\programs\python\python37\lib\site-packages\P
yInstaller\loader\pyimod03_importers.py", line 493, in exec_module
    exec(bytecode, module.__dict__)
  File "pyabf\__init__.py", line 14, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Sebu\\AppDat
a\\Local\\Temp\\_MEI89202\\pyabf\\version.txt'
[8900] Failed to execute script test
```

It shows that the pyABF module is crashing in line 14 in file `__init__.py`, where `open(version_path) ` crashes because the file `version.txt` does not exist. Apparently, pyInstaller does not export it.

### Available workaround

I figured out a workaround, but I really do not like it. Let me describe it and hopefully, you understand why I do not like it.

First, I created a folder within the root directory of the project. In our example, next to `test.py`, I created the folder `pyabf`. Into this folder, I copied the file that was missing: `version.txt`. The destination is located wherever the pyabf module is stored, for me it was `<pythonpath>/lib/site-packages/pyabf`. 

To create the executable and add the created folder into it, I executed `pyinstaller test.py --onefile --add-data "pyabf;pyabf"`. Running the executable finally gives the desired output:

```
hello
<class 'pyabf.abf.ABF'>
bye
```

### Now what?

I have a problem and a workaround. I do not see the workaround as a viable solution to the problem because it requires quite a lot of manual tweaking whenever the pyABF version updates. Also, it just feels very hacky.

To conclude, I ask you for your help. Did you happen to see this problem before and find a better workaround, or do you have any ideas on how to improve this? Anyway, this can help users in the future and could be added to the wiki / website.

Thanks a lot!

**Comment by swharden** [comment]
Hi @SebastianOltmanns, thanks for raising this issue!

At its core seems to be a problem with pyInstaller (not a pyabf bug), but I'm happy to do what I can to modify pyabf to make it easier to package! Getting fancy by looking-up version in Setup.py has caused problems in the past too (#104/#103), so eliminating this technical debt by hard-coding the version may be the way to go. 

Since pyABF is now very stable, manually incrementing version numbers won't be too laborious. For the record version numbers have to be updated in these two places:

https://github.com/swharden/pyABF/blob/9a8e7100e2a3a386370424112f542a573756f056/src/setup.py#L18

https://github.com/swharden/pyABF/blob/9a8e7100e2a3a386370424112f542a573756f056/src/pyabf/__init__.py#L15

I'll make this change and release on pypi shortly, then post a follow-up to let you know when it's ready to try out. Thanks for bringing this to my attention @SebastianOltmanns, and I'm happy to hear you're enjoying pyabf!

**Comment by SebastianOltmanns** [comment]
@swharden Thanks for helping on this. Im excited to see the new version and will give you feedback on how the packaging works once the new version is available.

**Comment by swharden** [comment]
I just released pyABF 2.2.8 with hard-coded version numbers. Let me know if it does the trick!

```
pip install --upgrade pyabf
```

**Comment by SebastianOltmanns** [comment]
It's working. Thanks again for the fast response and the fix!
